cloud/amazon/ec2_addresses.py

```diff
@@ -0,0 +1,409 @@
+        description:
+            - Interface ID, required for association EIP's to private IP's,
+                assigning more private ips, unassigning private ips.
+        required: false
+    state:
+        description:
+            - present to ensure resource is created or updated.
+            - absent to remove resource
+        required: false
+        default: present
+        choices: [ "present", "absent"]
+author: Karen Cheng(@Etherdaemon)
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Simple example of attaching 2 secondary private ips to
+# an interface. Returns a list of secondary addresses
+- name: Ensure interface has 2 seconday private addresses
+  ec2_addresses:
+    type: secondary_private
+    state: present
+    count: 2
+    interface_id: eni-12345678
+    profile: "{{ boto_profile }}"
+    region: "{{ region }}"
+  register: interface_results
+
+- name: Get the last secondary private ip address
+  set_fact:
+    last_secondary_private_details:
+      ip_address: "{{ interface_results.result.PrivateIpAddresses | selectattr('Primary', 'equalto', false) | map(attribute='PrivateIpAddress') | list | last }}"
+      interface_id: "{{ interface_results.result.NetworkInterfaceId }}"
+
+- name: Allocate Elastic IP and assign to the last secondary private ip address
+  ec2_addresses:
+    type: eip_to_private
+    state: present
+    private_ip: "{{ last_secondary_private_details.ip_address }}"
+    interface_id: "{{ last_secondary_private_details.interface_id }}"
+    profile: "{{ boto_profile }}"
+    region: "{{ region }}"
+    reassociation: true
+    state: present
+
+- name: Associate existing public ip to private address
+  ec2_addresses:
+    type: eip_to_private
+    state: present
+    public_ip: 52.1.1.1
+    private_ip: 10.1.1.1
+    interface_id: eni-12345678
+    state: present
+
+- name: Disassociate elastic ip from private address and release elastic ip
+  ec2_addresses:
+    type: eip_to_private
+    state: absent
+    public_ip: 52.1.1.1
+    release_on_disassociation: yes
+
+- name: Disassociate all elastic ips from interface and release elastic ips
+  ec2_addresses:
+    type: eip_to_private
+    state: absent
+    interface_id: eni-12345678
+    release_on_disassociation: yes
+
+- name: Unassign 10.1.1.1 from interface
+  ec2_addresses:
+    type: secondary_private
+    state: absent
+    private_ip: 10.1.1.1
+    interface_id: eni-12345678
+
+- name: Unassign all secondary addresses from interface
+  ec2_addresses:
+    type: secondary_private
+    state: absent
+    interface_id: eni-12345678
+
+'''
+
+try:
+    import json
+    import botocore
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
+
+
+def get_interface_details(client, module):
+    params = dict()
+    if not module.params.get('interface_id'):
+        module.fail_json(msg='interface_id is required')
```

Comment:
Should probably say in what circumstance interface_id is required (it's probably better to fail before getting to `get_interface_details` as it's not reflecting a function of the module)